packages/tooling/fast-tooling/src/data-utilities/shortcuts.spec.md

```diff
@@ -0,0 +1,119 @@
+# Abstract
+
+The shortcuts utility accepts an instance of the `MessageSystem` and through this sends information about available shortcuts and events.
+
+## Instantiating
+
+All the `Shortcuts` class needs to instantiate is an instance of the `MessageSystem`. It should then fire a custom event notifying all other items registered to the message system that shortcuts are available, with a listener that will capture keypress and identify which shortcut action to execute.
+
+```typescript
+import { Shortcut } from "@microsoft/fast-tooling"
+
+new Shortcuts({
+    messageSystem: fastMessageSystem
+});
+```
+
+## Adding shortcut actions
+
+Shortcut actions can either be custom or imported from a set of actions available.
+
+```typescript
+import { DuplicateShortcutAction, ShortcutAction, Shortcuts } from "@microsoft/fast-tooling"
+
+new Shortcuts({
+    messageSystem: fastMessageSystem,
+    actions: [
+        // Custom shortcut
+        new ShortcutAction({
```

Comment:
It would be interesting to consider adding a `matches` method to the `ShortcutAction` implementation that would take an `event` object as input and check whether its keys match the event. In that way, various consumers of these shortcuts wouldn't need to duplicate that kind of code or necessarily know the details of how things are triggered. They simply wire the indicated event and then check matches on the event object.

Reply:
That's a good point, I did something similar on the custom components that go into the `<Form />` react component.